### PR TITLE
Add fishing spot tooltips to worldmap

### LINKED ISSUE #7317: Show fishing spot type in world map
https://i.imgur.com/Dlb9irZ.png
Add what fish can be caught at a fishing spot when hovered over on the minimap.
Example. Fishing Spot - Lobster, Swordfish, Tuna
Could maybe add the needed fishing equipment as well.

### PR DESCRIPTION
**Pull request by melkypie**
Closes #7317

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/game/FishingSpot.java [hunk 1]
```diff
@@ -171,12 +171,13 @@
...
 		FISHING_SPOT_1536
 	),
-	COMMON_TENCH("Common tench, Bluegill, Greater siren, Mottled eel", ItemID.COMMON_TENCH,
+	COMMON_TENCH("Common tench, Bluegill, Greater siren, Mottled eel", "Common tench", ItemID.COMMON_TENCH,
 		FISHING_SPOT_8523);
 
 	private static final Map<Integer, FishingSpot> SPOTS;
 
 	private final String name;
+	private final String highestLevelFish;
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/game/FishingSpot.java:180]: This doesn't always represent the highest level fish, so either fix the entries for which that's not the case or change the name.

**Comment by dekvall** [comment]
You are also missing the frog spawn spots in the Lumbridge swamp caves.

**Comment by melkypie** [comment]
> You are also missing the frog spawn spots in the Lumbridge swamp caves.

Frog spawns are the same ones as the cave eel ones which are in there

**Comment by melkypie** [comment]
PR is ready for review

File: runelite-client/src/main/java/net/runelite/client/game/FishingSpot.java [hunk 2]
```diff
@@ -196,13 +197,19 @@
...
 
 	FishingSpot(String spot, int fishSpriteId, int... ids)
+	{
+		this(spot, spot, fishSpriteId, ids);
+	}
+
+	FishingSpot(String spot, String worldMapTooltip, int fishSpriteId, int... ids)
 	{
 		this.name = spot;
+		this.worldMapTooltip = worldMapTooltip;
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/game/FishingSpot.java:241]: The `worldMapTooltip` addition should be added in the same commit as FishingSpotLocation and the other world map code is added, not the one where `FishingSpot` is moved to a different package.

Comment (melkypie) [reply runelite-client/src/main/java/net/runelite/client/game/FishingSpot.java:241]: fixed the commits and rebased

**Comment by Nightfirecat** [comment]
The miscellania/etceteria dock fishing spot is still absent, though from what I can see, that is only useful for gaining approval among citizens of the island.

**Comment by melkypie** [comment]
> The miscellania/etceteria dock fishing spot is still absent, though from what I can see, that is only useful for gaining approval among citizens of the island.

Added etceteria dock as Lobster (approval only)

**Pull request merged**